userland/test/test_client.cpp

```diff
@@ -0,0 +1,84 @@
+// Copyright 2015 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <chrono>
+#include <iostream>
+#include <string>
+#include <thread>
+
+#include <rclcpp/rclcpp.hpp>
+
+#include <userland_msgs/AddTwoInts.h>
+
+template<typename T>
+int request(
+  rclcpp::Node::SharedPtr node,
+  std::function<void(typename T::Request::Ptr &, size_t)> set_data_func,
+  size_t number_of_requests)
+{
+  int rc = 1;
+  auto client = node->create_client<T>("service_name");
+  typename T::Request::Ptr ros_request(new typename T::Request());
+
+  auto start = std::chrono::steady_clock::now();
+  size_t i = 1;
+  while (rclcpp::ok() && i <= number_of_requests) {
+    set_data_func(ros_request, i);
+    auto f = client->async_send_request(ros_request);
+    std::cout << "requested service #" << i << std::endl;
+
+    auto wait_for_response_until = std::chrono::steady_clock::now() +
+      std::chrono::seconds(1);
+    std::future_status status;
```

Comment:
You can cherry pick https://github.com/ros2/rclcpp/pull/20/files#diff-690567624ee368716009118718985564R81 and https://github.com/ros2/rclcpp/pull/20/files#diff-294cbb523fb3d56a1df2a0a7ff0beb9bR31 to replace this snippet with `spin_until_future_complete`